packages/vite/src/node/config.ts

```diff
@@ -932,6 +932,7 @@ async function bundleConfigFile(
     entryPoints: [fileName],
     outfile: 'out.js',
     write: false,
+    target: ['node14.18', 'node16'],
```

Comment:
This make sense to me. Perhaps we can use `process.versions.node` here too to futureproof it.

Reply:
Won't using `process.versions.node` is a bit dangerous? Since it will cause vite to generate different bundles in different environments and become hard to debug.

Reply:
I think it might be fine as it's only syntax transform change, was hoping that `process.versions.node` reduces the work needed for esbuild to down-level the syntax if the user is using a higher version of node. I'm also fine with sticking with this PR though.

Reply:
Using `process.versions.node` will enable us to support non-transpilable features to work automatically (like BigInt).
But I think avoiding this is better for debuggability as @antfu said.
If we faced those features in future, we could add them to `supported` option. (I believe there won't be many.)